### PR TITLE
feat(network): Added ability to resolve DNS records for peers instead of only IP addresses

### PR DESCRIPTION
**Pull request by chefsale**
We use kuberenetes for spinning up networks now with custom genesis using DNs instead of IPs and we have a problem that nodes can't resolve DNS records which are assigned to services. This fixes the issue.

### CODE REVIEW

File: chain/network-primitives/src/network_protocol/mod.rs
Comment (pompon0) [inline chain/network-primitives/src/network_protocol/mod.rs:472]: Use Ipv4Addr::LOCALHOST (same for Ipv6)
https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html#associatedconstant.LOCALHOST

File: chain/network-primitives/src/network_protocol/mod.rs
Comment (pompon0) [inline chain/network-primitives/src/network_protocol/mod.rs:83]: while you are at it, would you mind adding a docstring to this function listing accepted formats?

Comment (pompon0) [reply chain/network-primitives/src/network_protocol/mod.rs:83]: also the following docstrings require an update:
- https://github.com/near/nearcore/blob/c39ed74feeeb43e27367e4a90f47540b5d361d59/nearcore/src/config.rs#L212
- (this one is missing the formats in the docstring, please add): https://github.com/near/nearcore/blob/c39ed74feeeb43e27367e4a90f47540b5d361d59/nearcore/src/config.rs#L206

**Comment by pompon0** [comment]
LGTM, thank you for your contribution. I've just left some minor comments.

**Review by pompon0 — APPROVED** [review]

File: chain/network-primitives/src/network_protocol/mod.rs
Comment (pompon0) [inline chain/network-primitives/src/network_protocol/mod.rs:469]: Technically, what "localhost" resolves to depends on /etc/hosts, so there is no strong guarantee that it will be 127.0.0.1. Please add a TODO to this test that it might require an improvement (probably by mocking the DNS resolution).

**Comment by chefsale** [comment]
@pompon0 @medicz has force pushed the updates. Can you please review it now.

**Review by jakmeier — APPROVED** [review]
LGTM, let's merge it. (It seems I cannot merge it because I cannot update the branch in your repo.)

**Comment by jakmeier** [comment]
@medicz your latest force-push is missing the changes in `nearcore/src/config.rs`. Can you please add that again?

(Adding new commits is usually easier for review, all commits will be squashed upon merging.)

**Comment by chefsale** [comment]
https://github.com/calimero-is-near/nearcore/commit/41bc437cfdfaded8890d150d735a18155ca300ce @medicz I pushed the change without the config.rs, can you please forcw push that change into our repo.

**Comment by chefsale** [comment]
Also, it is unclear to me how is responsible for landing this chamge in nearcore, as it says we dont have write access to merge it. 🤘

**Comment by medicz** [comment]
> @medicz your latest force-push is missing the changes in `nearcore/src/config.rs`. Can you please add that again?
> 
> (Adding new commits is usually easier for review, all commits will be squashed upon merging.)

Commit https://github.com/near/nearcore/commit/493aabf81155111c66ac6886e505e356f69998bd removed lines to which I added comments, and I did not realize it was moved to another file. Added now to that file.

**Comment by jakmeier** [comment]
> Also, it is unclear to me how is responsible for landing this change in nearcore, as it says we dont have write access to merge it. metal

I can press the squash-and-merge button when the branch is up-to-date. But yeah, I does seem suboptimal that two people have to sync in time here, someone for updating the branch and someone to actually merge.

**Pull request merged**
